extensions/common/context.cc

```diff
@@ -105,64 +102,58 @@ void getDestinationService(bool use_host_header, std::string* dest_svc_host,
                          RequestInfo* request_info) {
-  request_info->is_populated = true;
-  getValue({"cluster_name"}, &request_info->upstream_cluster);
-  getValue({"route_name"}, &request_info->route_name);
-  // Fill in request info.
-  // Get destination service name and host based on cluster name and host
-  // header.
-  getDestinationService(
-      use_host_header_fallback, &request_info->destination_service_host,
-      &request_info->destination_service_name, request_info->upstream_cluster,
-      request_info->route_name);
-
-  getValue({"request", "url_path"}, &request_info->request_url_path);
-
-  uint64_t destination_port = 0;
-  if (outbound) {
-    getValue({"upstream", "port"}, &destination_port);
-    getValue({"upstream", "uri_san_peer_certificate"},
-             &request_info->destination_principal);
-    getValue({"upstream", "uri_san_local_certificate"},
-             &request_info->source_principal);
-  } else {
-    getValue({"destination", "port"}, &destination_port);
-
-    bool mtls = false;
-    if (getValue({"connection", "mtls"}, &mtls)) {
-      request_info->service_auth_policy =
-          mtls ? ::Wasm::Common::ServiceAuthenticationPolicy::MutualTLS
-               : ::Wasm::Common::ServiceAuthenticationPolicy::None;
+  if (!request_info->is_populated) {
+    request_info->is_populated = true;
+
+    getValue({"cluster_name"}, &request_info->upstream_cluster);
+    getValue({"route_name"}, &request_info->route_name);
+    // Fill in request info.
+    // Get destination service name and host based on cluster name and host
+    // header.
+    populateDestinationService(use_host_header_fallback, request_info);
+    uint64_t destination_port = 0;
+    if (outbound) {
+      getValue({"upstream", "port"}, &destination_port);
+      getValue({"upstream", "uri_san_peer_certificate"},
+               &request_info->destination_principal);
+      getValue({"upstream", "uri_san_local_certificate"},
+               &request_info->source_principal);
+    } else {
+      getValue({"destination", "port"}, &destination_port);
+
+      bool mtls = false;
+      if (getValue({"connection", "mtls"}, &mtls)) {
+        request_info->service_auth_policy =
+            mtls ? ::Wasm::Common::ServiceAuthenticationPolicy::MutualTLS
+                 : ::Wasm::Common::ServiceAuthenticationPolicy::None;
+      }
+      getValue({"connection", "uri_san_local_certificate"},
+               &request_info->destination_principal);
+      getValue({"connection", "uri_san_peer_certificate"},
+               &request_info->source_principal);
     }
-    getValue({"connection", "uri_san_local_certificate"},
-             &request_info->destination_principal);
-    getValue({"connection", "uri_san_peer_certificate"},
-             &request_info->source_principal);
+    request_info->destination_port = destination_port;
   }
-  request_info->destination_port = destination_port;
-
-  uint64_t response_flags = 0;
```

Comment:
why remove from here?

Reply:
This is called repeatedly in report(). The first time it's called, response flag is going to be 0, and be set to 0 till the end.

Reply:
BTW I moved it both TCP and HTTP populate calls instead.